### DISCUSSION TITLE
Welcome to HASS-plate-recognizer Discussions!

### DISCUSSION THREAD

**Discussion by robmarkcole** [robmarkcole/HASS-plate-recognizer#24] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by healeydave** [comment]
Hi, 
I'm Dave, I would just like thank @robmarkcole for his excellent work on this project.

Initially, I thought the module wasn't working because I was getting some errors in Home Assistant and not noticing many pictures in the "plates" folder on my server but it turns out it was working perfectly when I signed into platerecognizer and found lots of images on there. [embarrassing]

Whilst I'm not getting much success at the moment, I think that's down to my own setup and trying to use an exiting camera (and mounting position) that was not originally set-up with vehicle detection in mind.

I'm currently trying to fine tune my settings to see if I can improve things but I have resigned myself to the idea that I may well end up having to install a new (and better positioned) camera.

I have just tried switching from basic motion event triggering to smart events using line crossing a>b to see if that helps.

Thanks...

**Reply by healeydave** [reply]
Just thought I'd post an update:
I really want this integration to work with HASS so I bought a new camera. A Hikvision DS-2CD2785G0 4K (8MP) and set it up at the front gate, so it can capture vehicles as they slow down to enter the gate.

For some reason (I cannot figure out why), all I'm getting posted to plate organiser is a lovely picture of the scene with no vehicle in the frame at all :(
Deeply disappointing and frustrating. I have played around with the settings for hours. switching between Line Crossing and Intrustion Detection, moving area's, changing sensitivities, nothing works. I can only assume the time it's taking the processing to act capture is so slow, the vehicle has long gone out of the frame. I can't even get close to getting even the rear of a vehicle in shot.

When I started out with the original Hikvision camera further down the drive, I could get the vehicles in shot and I played around with the settings on that camera originally and I did manage to get the best angle I could but its simply wasn't good enough for recognition. So I'm totally stumped as to why I can't get this new camera to capture a vehicle at all.

I have had success with Blue Iris. I found out that BI also supports plate organiser so I configured the new camera in BI and that manages to capture the vehicles in a timely fashion and recognises their plates due to the better camera position and the better quality images produced on this camera but of-course Blue Iris uses its own detection processing, not the camera's.

I'm gutted I can't get this integration to work because I don't really want to have another system running Blue Iris just to capture the vehicle plates.

**Reply by robmarkcole** [reply]
Hi @healeydave 
It sounds like your motion trigger delay is too much, and the car it out of shot by the time scanning is performed. Presumably you have some video captures so I recommend extracting a frame from the video with the car clearly in shot, and processing that just gain confidence in the processing capability. 
Since motion detection is notoriously difficult to fine tune, you perhaps should consider switching to an object based trigger, e.g. using deepstack to detect when the car is in frame and then processing.

I wrote an article on this topic on https://www.hackster.io/robin-cole/add-ai-brains-to-your-home-camera-system-5195b0

**Reply by healeydave** [reply]
I don't think there is a setting in the Hikvision NVR to delay the motion. I'm using "Line Crossing" and You just draw a line from A to B and set the sensitivity. I have put the sensitivity on high and the line is right near the gate entrance. The system is working, I'm getting no false positives, but the I check Plate recogniser, the photo contains no car. I can't understand where the delay is happening. Its like the photo that gets sent to plate recogniser is taken several seconds later than it should be, but I can't move the line crossing any further away to make it activate any sooner.

I looked at the article, I realised we talked about this before, I don't think my HA is going to be powerful enough to do all that. I have a 4GB RPi4 with an Argon case and SSD drive. I can't afford to replace all that at the moment.

**Reply by robmarkcole** [reply]
Yeah there will be some delays. I think calling the image_processing `scan` service may even capture a new image. In general image processing in home assistant is not 'real time', so you may need a faster solution like [frigate](https://github.com/blakeblackshear/frigate) to help with the captures

**Reply by healeydave** [reply]
I do have Frigate NVR installed with a USB Coral.AI, that does a good job of detecting objects like cars but unfortunately, it doesn't do number plates at the moment. I'm hoping its something the developer is working on.

**Comment by healeydave** [comment]
This is the Hikvision settings:

<img width="581" alt="Screenshot 2021-03-25 at 15 51 10" src="https://user-images.githubusercontent.com/5435141/112502521-0d19ed80-8d82-11eb-8bfe-10e02242adc8.png">


and this is the photo uploaded to plate recogniser:

![1528_3pDOy_upload](https://user-images.githubusercontent.com/5435141/112502751-3cc8f580-8d82-11eb-8e71-e5baefe9c3b9.jpg)

**Reply by healeydave** [reply]
As you can see its not even close to capturing  vehicle :)

**Reply by robmarkcole** [reply]
Probably the forums is the best place to go for Hikvision settings advice

**Comment by healeydave** [comment]
This is Frigate's capture at exactly the same time:

https://user-images.githubusercontent.com/5435141/112503771-1ce60180-8d83-11eb-82a6-8c6de1335865.mp4

**Reply by robmarkcole** [reply]
You want to run that through a number plate detection model, then send the frame with highest confidence to plate recognizer
